src/app/address/controllers/addressResults.js

```diff
@@ -14,22 +14,15 @@ class AddressResultsController extends BaseController {
         const selectedAddress = req.body["address-selection"];
         const allAddresses = req.sessionModel.get("searchResults");
 
-        const choosenAddress = allAddresses.find(
-          (address) => address.text === selectedAddress
-        );
+        const chosenAddress = Object.assign(
+          {},
+          allAddresses.find((address) => address.text === selectedAddress)
+        ); // force deep copy
```

Comment:
this fixes a bug I found whilst navigating backwards from the confirmation page back to the address results where we were modifying the chosen element in the array